File: src/main/java/org/zenbot/szolnok/timetable/batch/domain/BusStopWithBuses.java
package org.zenbot.szolnok.timetable.batch.domain;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Set;

@Data
@Document
@NoArgsConstructor
@AllArgsConstructor
public class BusStopWithBuses {

    @Id
    private String id;

    private String busStopName;
    private Set<String> buses;
}

File: src/main/kotlin/org/zenbot/szolnok/timetable/domain/BusStopWithBuses.kt
package org.zenbot.szolnok.timetable.domain

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document
data class BusStopWithBuses(
    @Id var id: String,
    var busStopName: String,
    var buses: Set<String>
)